slice_experiments broken if image_range doesn't start at 1

```
def test_slice_experiments_centroid_test_data_starting_from_2(dials_data):
    files = dials_data("centroid_test_data").listdir("*.cbf", sort=True)[1:]
    experiments = ExperimentListFactory.from_filenames(f.strpath for f in files)
    sliced_image_range = [(2, 4)]
    sliced_experiments = slice_experiments(experiments, sliced_image_range)
    assert sliced_experiments[0].scan.get_image_range() == sliced_image_range[0]
```
gives the following error:
```
util/slice.py:37: in slice_experiments
    exp.imageset = exp.imageset[beg:end]
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <dxtbx_imageset_ext.ImageSequence object at 0x119c9b110>, item = slice(0, 3, None)

    def __getitem__(self, item):
        """Get an item from the sequence stream.
    
        If the item is an index, read and return the image at the given index.
        Otherwise, if the item is a slice, then create a new Sequence object
        with the given number of array indices from the slice.
    
        Params:
            item The index or slice
    
        Returns:
            An image or new Sequence object
    
        """
        if isinstance(item, slice):
            offset = self.get_scan().get_batch_offset()
            if item.step is not None:
                raise IndexError("Sequences must be sequential")
    
            # nasty workaround for https://github.com/dials/dials/issues/1153
            # slices with -1 in them are meaningful :-/ so grab the original
            # constructor arguments of the slice object.
            # item.start and item.stop may have been compromised at this point.
            if offset < 0:
                start, stop, step = item.__reduce__()[1]
                if start is None:
                    start = 0
                else:
                    start -= offset
                if stop is None:
                    stop = len(self)
                else:
                    stop -= offset
                return self.partial_set(start, stop)
            else:
                start = item.start or 0
                stop = item.stop or (len(self) + offset)
>               return self.partial_set(start - offset, stop - offset)
E               OverflowError: can't convert negative value to unsigned int

../dxtbx/imageset.py:264: OverflowError
```

--- a/newsfragments/1383.bugfix
+++ b/newsfragments/1383.bugfix
@@ -1,0 +1,1 @@
+Fix for slice_experiments if image range doesn't start at 1

--- a/util/slice.py
+++ b/util/slice.py
@@ -34,7 +34,10 @@ def slice_experiments(experiments, image_ranges):
         end = sr[1] - arr_start
         exp.scan.swap(exp.scan[beg:end])
         if exp.imageset is not None:
-            exp.imageset = exp.imageset[beg:end]
+            # Gorilla of temporary workarounds for inconsistent scan and imageset slicing
+            # https://github.com/cctbx/dxtbx/issues/213
+            offset = exp.scan.get_batch_offset()
+            exp.imageset = exp.imageset[beg + offset : end + offset]
 
         # account for scan-varying crystal
         if exp.crystal and exp.crystal.num_scan_points > 0:

--- a/util/test_slice.py
+++ b/util/test_slice.py
@@ -1,7 +1,6 @@
 from __future__ import absolute_import, division, print_function
 
 import copy
-import pytest
 
 from dxtbx.model import Experiment, ExperimentList
 from dxtbx.model import Scan
@@ -32,9 +31,6 @@ def test_slice_experiments_centroid_test_data(dials_data):
     assert copy.deepcopy(sliced_experiments)
 
 
-@pytest.mark.xfail(
-    raises=OverflowError, reason="https://github.com/dials/dials/issues/1382"
-)
 def test_slice_experiments_centroid_test_data_starting_from_2(dials_data):
     files = dials_data("centroid_test_data").listdir("*.cbf", sort=True)[1:]
     experiments = ExperimentListFactory.from_filenames(f.strpath for f in files)

